### PR TITLE
TF Hub based classifier

### PR DESCRIPTION
**Pull request by sorensenjs**
This is a branch of the gru model using the TF Hub sentence encoder.

My quick experiments suggest that this is highly competitive with our existing models.
Achieving 0.985 AUC

This pulls out some implementation classes from common into run.py, but this can be
potentially addressed going forward by generalizing the common code.

### CODE REVIEW

File: experiments/tf_trainer/tf_hub_classifier/model.py [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+    'https://tfhub.dev/google/universal-sentence-encoder/2',
+    'The url of the TF Hub sentence encoding module to use.')
+# This would normally just be a multi_integer, but we use string due to
+# constraints with ML Engine hyperparameter tuning.
+tf.app.flags.DEFINE_string(
+    'dense_units', '512,128,64',
+    'Comma delimited string for the number of hidden units in the dense layer.')
+
+
+class TFRNNModel(base_model.BaseModel):
```
Comment (jjtan) [inline experiments/tf_trainer/tf_hub_classifier/model.py]: Lets correct the class name. Maybe TFHubClassifierModel

File: experiments/tf_trainer/tf_hub_classifier/model.py [hunk 2]
```diff
@@ -0,0 +1,83 @@
+"""Tensorflow Estimator implementation of RNN Model with Attention"""
```
Comment (jjtan) [inline experiments/tf_trainer/tf_hub_classifier/model.py]:
```suggestion
"""Simple TF Hub based model using the Universal Sentence Encoder and basic Dense layers."""
```

File: experiments/tf_trainer/tf_hub_classifier/run.py [hunk 3]
```diff
@@ -0,0 +1,164 @@
...
+                            "The number of steps to train for.")
+tf.app.flags.DEFINE_integer("eval_period", 500,
+                            "The number of steps per eval period.")
+tf.app.flags.DEFINE_integer("eval_steps", 50,
+                            "The number of steps to eval for.")
+
+# TODO: Missing fields are not handled properly yet.
+LABELS = {
+    "frac_neg": tf.float32,
+    #"frac_very_neg": tf.float32
```
Comment (jjtan) [inline experiments/tf_trainer/tf_hub_classifier/run.py]: might as well remove this

File: experiments/tf_trainer/tf_hub_classifier/run.py [hunk 4]
```diff
@@ -0,0 +1,164 @@
...
+}  # type: Dict[str, tf.DType]
+
+
+class TFSimpleRecordInput(dataset_input.DatasetInput):
```
Comment (jjtan) [inline experiments/tf_trainer/tf_hub_classifier/run.py]: Optional, but I'd say this deserves it's own class in common and then next step can be a refactoring of the existing input class to use it.

File: experiments/tf_trainer/tf_hub_classifier/model.py (hunk 1 above)
Comment (sorensenjs) [reply experiments/tf_trainer/tf_hub_classifier/model.py]: Done.

File: experiments/tf_trainer/tf_hub_classifier/model.py (hunk 2 above)
Comment (sorensenjs) [reply experiments/tf_trainer/tf_hub_classifier/model.py]: Done.

File: experiments/tf_trainer/tf_hub_classifier/run.py (hunk 3 above)
Comment (sorensenjs) [reply experiments/tf_trainer/tf_hub_classifier/run.py]: Done.

File: experiments/tf_trainer/tf_hub_classifier/run.py (hunk 4 above)
Comment (sorensenjs) [reply experiments/tf_trainer/tf_hub_classifier/run.py]: Done.

**Comment by sorensenjs** [comment]
Successfully ran on cloud. Submitting (or "Merging" or whatever...)

**Pull request merged**